src/geometry/ArborX_Ray.hpp

```diff
@@ -40,8 +40,13 @@ struct Vector : private Point
   }
 };
 
-KOKKOS_INLINE_FUNCTION constexpr Vector makeVector(Point const &begin,
-                                                   Point const &end)
+template <typename Point1, typename Point2>
+KOKKOS_INLINE_FUNCTION constexpr std::enable_if_t<
+    GeometryTraits::is_point<Point1>{} && GeometryTraits::is_point<Point2>{} &&
```

Comment:
I don't quite understand why there are two `Point` template parameters. Are there times when they are different?

And I assume we need to template to be able to construct `Vector` from both `ArborX::Point` and `ArborX::ExperimentalHyperGeometry::Point`?

Reply:
Yes, with the current code, we have cases where the first argument comes from the non-hyper classes and the second one does. Thus, allowing to mix arbitrarily between the two seems more sensible than to create four separate overloads.

Reply:
Could you speak for when the two types are different? I'm trying to understand this.

Reply:
Yes.